packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -487,26 +506,25 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     const id = this._id;
     const isItemSelected =
       item.index !== undefined && selectedIndices ? selectedIndices.indexOf(item.index) > -1 : false;
-    const checkboxStyles = () => {
-      return getCheckboxStyles({
-        theme: getTheme(),
-        checked: isItemSelected,
-        disabled: item.disabled
-      });
-    };
+
+    const itemClassName =
```

Comment:
we could also move this logic to the styles file, might make more sense there? dunno.

Reply:
It would perhaps be more intuitive but this logic varies for each option - unless we pass in an array of selected/disabled combination as an array in the styleProps......... (which I didn't initially consider doing).

Let me know if you think this is what I should do...

Reply:
This approach however is probably much simpler... the alternative of passing in the array means we have to compute the render state of every option at the very start of the main render function - and very likely doing the same computation twice (once before getClassNames and another time once more in the actual render for using it in other props...).

Reply:
A 3rd option is just to break apart the Dropdown component into Dropdown and DropdownOptions but that is a refactor that I don't feel has good ROI at this point in time.